virttest/utils_zchannels.py

```diff
@@ -0,0 +1,168 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2020
+# Author: Sebastian Mitterle <smitterl@redhat.com>
+
+"""
+Module for channel device management on s390x aka IBM Z
+
+Example:
+    # Retrieve device to remove it from host
+
+    subchannel_paths = SubchannelPaths()
+    subchannel_paths.get_info()
+
+    device = subchannel_paths.get_first_unused_and_safely_removable()
+    chpids = device[subchannel_paths.header["CHPIDs"]]
+
+    channel_paths = ChannelPaths()
+    channel_paths.set_standby(chipds)
+"""
+from virttest.utils_misc import cmd_status_output
+
+# timeout value in seconds for any command run in this module
+CMD_TIMEOUT = 60
+# for debugging, set True to get verbose command run output
+VERBOSE = False
+
+
+class SubchannelPaths(object):
+    """
+    Represents channel subsystem info
+    """
+
+    def __init__(self, session=None):
+        """
+        Initializes instance
+
+        :param session: guest session; if None, host info is handled
+        """
+        self.session = session
+        self.header = {
```

Comment:
Could header is  Class-wide  constant since it looks like its values are hardcode?

Reply:
@chunfuwen It could. I updated it. But you'll see that in this case, IMO, it's only syntactic sugar, we still need to use `self.HEADER` or the longer `SubchannelPaths.HEADER` to use it; using only `HEADER` will lead to an error. However, as it's conceptually true - the header can be expected to be constant - I updated it and re-ran the unittests.